spring-boot-project/spring-boot/src/main/java/org/springframework/boot/convert/PeriodStyle.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright 2002-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.convert;
+
+import java.time.Period;
+import java.time.temporal.ChronoUnit;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * A standard set of {@link Period} units.
+ *
+ * @author Eddú Meléndez
+ * @author Edson Chávez
+ * @since 2.3.0
+ * @see Period
+ */
+public enum PeriodStyle {
+
+	/**
+	 * Simple formatting, for example '1d'.
+	 */
+	SIMPLE("^([\\+\\-]?\\d+)([a-zA-Z]{0,2})$") {
```

Comment:
I'm happy to see this! My little grain of sand.

You don't need to escape `+` and `-` (if it's at the end) in a character class
```
SIMPLE("^([+-]?\\d+)([a-zA-Z]{0,2})$")
```

Reply:
That was accident that came about because I copy/pasted some code from `Period` :)

Reply:
I'll apply the same fix to the other patterns that we define.